src/mame/drivers/olyboss.cpp

```diff
@@ -0,0 +1,339 @@
+// license:BSD-3-Clause
+// copyright-holders: Gabriele D'Antona
+/* 
+	Olympia BOSS
+	Made in Germany around 1981
+	
+	The BOSS series was not a great success, as its members differed too much to be compatible:
+	First they were 8085 based, later machines used a Z80A.
+	
+	Other distinguishing features were the capacity of the disk drives:
+	
+	BOSS A: Two 128K floppy drives
+	BOSS B: Two 256K disk drives
+	BOSS C: Two 600K disk drives
+	BOSS D: One 600K disk drive, one 5 MB harddisk
+	BOSS M: M for multipost, up to four BOSS machines linked together for up to 20MB shared harddisk space
+	
+	Olympia favoured the French Prologue operating system over CPM (cf. Olympia People PC) and supplied BAL
+	as a programming language with it.
+		
+	Video is 80x28
+	
+	There are no service manuals available (or no documentation in general), so everything is guesswork.
+
+	- Ports 0x80 and 0x81 seem to be related to the graphics chip and cursor position
+	The rom outs value 0x81 to port 0x81 and then the sequence <column> <row> (?) to port 0x80
+
+	- The machine boots up and shows "BOSS .." on the screen. Every keystroke is repeated on screen.
+	If you press <return>, the machine seems to go into a boot sequence (from the HD, probably)
+	
+	The harddisk controller is based on a MSC-9056.
+	
+	Links: http://www.old-computers.com/museum/computer.asp?c=95
+*/
+
+#include "emu.h"
+#include "cpu/z80/z80.h"
+#include "machine/keyboard.h"
+#include "video/upd3301.h"
+#include "machine/i8257.h"
+#include "screen.h"
+
+#define Z80_TAG         "z80"
+#define UPD3301_TAG     "upd3301"
+#define I8257_TAG       "i8257"
+#define SCREEN_TAG		"screen"
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+class olyboss_state : public driver_device
+{
+public:
+	olyboss_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag),
+			m_maincpu(*this, Z80_TAG),
+			m_dma(*this, I8257_TAG),
+			m_crtc(*this,UPD3301_TAG),
+			m_char_rom(*this, UPD3301_TAG)
+		{ }
+
+	required_device<cpu_device> m_maincpu;
+	required_device<i8257_device> m_dma;
+	required_device<upd3301_device> m_crtc;
+	required_memory_region m_char_rom;
+
+	DECLARE_DRIVER_INIT(olyboss);
+	//uint32_t screen_update_olyboss(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
+
+	DECLARE_READ8_MEMBER( videoram_read );
+	DECLARE_WRITE8_MEMBER( videoram_write );
+
+	DECLARE_READ8_MEMBER(keyboard_read);
+
+	DECLARE_READ8_MEMBER(port_read);
+	DECLARE_WRITE8_MEMBER(port_write);
+
+	void keyboard_put(u8 data);
+
+	UPD3301_DRAW_CHARACTER_MEMBER( olyboss_display_pixels );
+
+	DECLARE_WRITE_LINE_MEMBER( hrq_w );
+	DECLARE_READ8_MEMBER( dma_mem_r );
+	
+	void olybossd(machine_config &config);
+
+private:
+
+	bool keybHit;
+	u8 keyStroke;
+	uint8_t m_mainVideoram[0x2000];
```

Comment:
There's inconsistent style here - you have one member with an `m_` prefix and two without.  Also, the object finder members above don't need to be public.